src/DependencyInjection/Compiler/RegisterFragmentsPass.php

```diff
@@ -77,6 +78,10 @@ protected function registerFragments(ContainerBuilder $container, string $tag):
                     $preHandlers[$identifier] = $reference;
                 }
 
+                if (is_a($definition->getClass(), AbstractFragmentController::class, true)) {
```

Comment:
That just kills the possibility to have a custom controller that needs to know the fragment options but does not want to extend the `AbstractFragmentController`. I mean, the purpose of the fragments is that they are fully independent controllers so I think we should work with a `FragmentOptionsAwareInterface` here, wdyt?

Reply:
I second that. We should use small(!) interfaces for these tasks.
Brings also the benefit of being able to deprecate one way (old interface) and introduce a new way without changing the API. I guess we will have to alter the mechanics in the future which is easier by adding another interface than having version_compares() and the like.